src/mscorlib/src/System/ArraySegment.cs

```diff
@@ -27,6 +27,12 @@ namespace System
     [Serializable]
     public struct ArraySegment<T> : IList<T>, IReadOnlyList<T>
     {
+        // Do not replace this with Array.Empty. We don't want to have the overhead of instantiating
+        // another generic type in addition to ArraySegment<T> for new type parameters.
+        private static readonly T[] s_emptyArray = new T[0];
+
+        public static ArraySegment<T> Empty { get; } = new ArraySegment<T>(s_emptyArray);
```

Comment:
Not sure if it would be better to do `new ArraySegment<T>(new T[0])` here and then return `Empty._array` in `ToArray()` to avoid another static field?